cli/src/torii/routing.rs

```diff
@@ -232,15 +202,18 @@ async fn handle_pending_transactions(
     sumeragi: SumeragiHandle,
     pagination: Pagination,
 ) -> Result<Scale<Vec<VersionedSignedTransaction>>> {
+    // TODO: Don't clone wsv here
     let wsv = sumeragi.wsv_clone();
-    Ok(Scale(
-        queue
-            .all_transactions(&wsv)
-            .into_iter()
-            .map(Into::into)
-            .paginate(pagination)
-            .collect::<Vec<_>>(),
-    ))
+
+    let query_response = queue
+        .all_transactions(&wsv)
+        .map(Into::into)
+        .paginate(pagination)
+        .collect::<Vec<_>>();
+    // TODO:
+    //.batched(fetch_size)
```

Comment:
I don't understand when it should be done. Consider adding description or create an issue

Reply:
#3742